pkg/controller/tf/controller.go

```diff
@@ -174,6 +174,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (res
 	if err != nil {
 		return reconcile.Result{}, fmt.Errorf("could not parse resource %s: %v", req.NamespacedName.String(), err)
 	}
+	if err := r.handleDefault(resource); err != nil {
```

Comment:
IIUC `handleDefault` is only handling default for state-into-spec. So shall we name it `handleDefaultStateIntoSpecValue` instead?

Reply:
Done.

Reply:
I quite liked handleDefault because it hinted that we can use this approach to eliminate all the mutating webhooks, but ... not a blocker.

Reply:
We can always generalize the function name. And as you suggested in https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1151#discussion_r1472951138, I'll do some refactoring around the defaulting logic.

So I'll keep the code as is in this PR.